lib/functions/RegularFunctions.ts

```diff
@@ -94,6 +94,10 @@ const equality = {
   overloads: declare(C.RegularOperator.EQUAL)
     .numberTest(() => (left, right) => left === right)
     .stringTest(() => (left, right) => left.localeCompare(right) === 0)
+    .set(
+      [ 'literal', 'literal' ],
+      () => ([ left, right ]) => bool(RDFTermEqual(left, right)),
+    )
```

Comment:
It would be more something like this. Note that this is more pseudocode than anything else :)

```suggestion
    .set(
      [ TypeUrl.LanguageString, TypeUrl.LanguageString ],
      () => ([ left, right ]) => bool(left.str() === right.str() && left.language() === right.language()),
    )
```

Reply:
Thanks, I will have look, I understand more now

Reply:
What do you think of the current code? I still forward the checking to RDFTermEqual. It would then pass to something like https://github.com/rdfjs/N3.js/blob/main/src/N3DataFactory.js#L111

Reply:
Ahh I see it goes through this one:

https://github.com/rubensworks/rdf-data-factory.js/blob/master/lib/Literal.ts#L33

Same idea.